working/base-interface-final/feature-specification.md

```diff
@@ -0,0 +1,470 @@
+# "Base", "interface", and "final" types
+
+Author: Bob Nystrom
+
+Status: In-progress
+
+Version 1.0
+
+This proposal specifies three modifiers that can be placed classes and mixins to
+allow an author to control whether the type allows being implemented, extended,
+both, or neither. The proposed modifiers are:
+
+*   No modifier: As today, the type has no restrictions.
+*   `base`: The type can be extended or mixed in but not implemented.
+*   `interface`: The type can be implemented but not extended or mixed in.
```

Comment:
The `interface class` can only be implemented. It cannot be combined with `base` or `mixin`.
Could we allow omitting the `class` and just write `interface Foo`?

Reply:
> Could we allow writing just interface to mean the same as abstract interface class?

I've toyed with that idea as I've noodled on these various proposal over the years, but I've generally leaned away from it because I suspect it might be confusing that a thing labeled only `interface` can still have static methods, constructors, and even concrete instance methods.

I figure the best path is to just have a handful of relatively simple modifiers and let them be composed in obvious ways without offering too many shorthands and sugared combinations.